core/src/main/java/org/apache/calcite/sql/SqlPeriodTypeNameSpec.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.util.Litmus;
+
+import static org.apache.calcite.util.Static.RESOURCE;
+
+/**
+ * A sql type name specification of the PERIOD type.
+ */
+public class SqlPeriodTypeNameSpec extends SqlTypeNameSpec{
+
+  public final TimeScale timeScale;
+  public final Integer precision;
+  public final boolean isWithTimezone;
+
+  public SqlPeriodTypeNameSpec(TimeScale timeScale,
+      SqlNumericLiteral precision,
+      boolean isWithTimezone, SqlParserPos pos) {
+    super(new SqlIdentifier("Period",pos), pos);
+
+    // date period cannot contain precision or with time zone token
```

Comment:
nit: punctuation.

Reply:
nit: and capitalization.

Reply:
Changed in https://github.com/googleinterns/calcite/pull/177/commits/110d83e563426a9ea05113263c6efc02c44447dc